lib/Doctrine/ORM/Configuration.php

```diff
@@ -83,12 +100,32 @@ public function getAutoGenerateProxyClasses()
     public function setAutoGenerateProxyClasses($autoGenerate)
     {
         $this->attributes['autoGenerateProxyClasses'] = (int) $autoGenerate;
+
+        $proxyManagerConfig = $this->getProxyManagerConfiguration();
+
+        switch ((int) $autoGenerate) {
+            case StaticProxyFactory::AUTOGENERATE_ALWAYS:
+            case StaticProxyFactory::AUTOGENERATE_FILE_NOT_EXISTS:
+                $proxyManagerConfig->setGeneratorStrategy(new FileWriterGeneratorStrategy(
+                    new FileLocator($proxyManagerConfig->getProxiesTargetDir())
+                ));
+
+                return;
+            case StaticProxyFactory::AUTOGENERATE_NEVER:
+            case StaticProxyFactory::AUTOGENERATE_EVAL:
+            default:
+                $proxyManagerConfig->setGeneratorStrategy(new EvaluatingGeneratorStrategy());
+
+                return;
+        }
     }
 
     /**
      * Gets the namespace where proxy classes reside.
      *
      * @return string|null
+     *
+     * @deprecated please do not use this anymore
```

Comment:
Should we drop this?

Reply:
Not removed: too commonly used, therefore keeping it and documented as deprecated